### PR TITLE
Added Whitelist methods, moved MaxPlayers to SpoutConfiguration, moved W...

### PR DESCRIPTION
**Pull request by dredhorse**
...hitelisting into BanManager

Needs: https://github.com/SpoutDev/SpoutAPI/pull/323

Signed-off-by: dredhorse dredhorse@breiden.net

### CODE REVIEW

File: src/main/java/org/spout/engine/SpoutServer.java [hunk 1]
```diff
@@ -103,6 +100,9 @@
 
 	public SpoutServer() {
 		this.filesystem = new ServerFileSystem();
+		instance = this;
+		whitelist =  SpoutConfiguration.USE_WHITELIST.getBoolean();
```
Comment (olliestanley) [inline src/main/java/org/spout/engine/SpoutServer.java]: There's a double space here :P

Comment (dredhorse) [inline src/main/java/org/spout/engine/SpoutServer.java]: see, I told you... now 2 "typos" any other comment about this pull request?

File: src/main/java/org/spout/engine/SpoutConfiguration.java [hunk 2]
```diff
@@ -36,15 +35,14 @@
...
-	public static final ConfigurationHolder BANLIST = new ConfigurationHolder(Arrays.asList("Satan"), "banlist");
 	public static final ConfigurationHolder CONSOLE_TYPE = new ConfigurationHolder("jline", "console");
 	public static final ConfigurationHolder DEFAULT_WORLD = new ConfigurationHolder("world", "default-world");
 	public static final ConfigurationHolder USE_WHITELIST = new ConfigurationHolder(false, "use-whitelist");
 	public static final ConfigurationHolder VIEW_DISTANCE = new ConfigurationHolder(10, "view-distance");
-	public static final ConfigurationHolder WHITELIST = new ConfigurationHolder(Arrays.asList("Notch", "ez", "jeb"), "whitelist");
 	public static final ConfigurationHolder WORLDS = new ConfigurationHolder("default", "worlds");
 	public static final ConfigurationHolder LIVE_LIGHTING = new ConfigurationHolder(false, "live-lighting");
 	public static final ConfigurationHolder LIGHTING_ENABLED = new ConfigurationHolder(true, "lighting-enabled");
+	public static final ConfigurationHolder MAX_PLAYERS = new ConfigurationHolder(20, "maxPlayers");
```
Comment (olliestanley) [inline src/main/java/org/spout/engine/SpoutConfiguration.java]: Yeah I have another comment. This is camelCase where the other things are words separated by dash. see live-lighting as a pose to maxPlayers.

File: src/main/java/org/spout/engine/command/BanCommands.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
+/*
+ * This file is part of spout (http://www.spout.org/).
+ *
+ * spout is licensed under the SpoutDev License Version 1.
+ *
```
Comment (olliestanley) [inline src/main/java/org/spout/engine/command/BanCommands.java]: I think Spout should be with a capital S. Also the website shouldn't be at the top, see the header.txt file.

File: src/main/java/org/spout/engine/command/BanNestedCommands.java [hunk 4]
```diff
@@ -0,0 +1,220 @@
+/*
+ * This file is part of spout (http://www.spout.org/).
```
Comment (olliestanley) [inline src/main/java/org/spout/engine/command/BanNestedCommands.java]: Same here as above file.

**Comment by olliestanley** [comment]
I think that's it :)

File: src/main/java/org/spout/engine/SpoutConfiguration.java (hunk 2 above)
Comment (dredhorse) [inline src/main/java/org/spout/engine/SpoutConfiguration.java]: thanks...

**Comment by dredhorse** [comment]
grammar nazi :-P

ok, any feedback from OTHERs and not about formatting? :-)

**Comment by Spouty** [comment]
**[AUTO-GENERATED COMMENT]**

This pull request has only **_1 commit**_ and is **_signed off**_.
Thanks for following the [pull request guidelines](http://spout.in/prguide) and for your contribution!
A [team member](http://spout.org/team) will review this pull request when time allows.

**Comment by robxu9** [comment]
JFYI, for license headers, you can run mvn clean ;)

**Comment by dredhorse** [comment]
yeah, I removed the automatic licensing in idea

**Comment by macalinao** [comment]
`mvn` works fine too

File: .gitignore [hunk 5]
```diff
@@ -71,4 +71,5 @@ nb-configuration.xml
 /plugins
 /resources
 /update
-/worlds
\ No newline at end of file
```
Comment (stephenwilliams) [inline .gitignore:74]: Why are you changing .gitignore in a PR for whitelisting?

Comment (dredhorse) [inline .gitignore:74]: I added an exclusion for the altlassian plugin... hmm did I really delete /worlds ?

Comment (dredhorse) [inline .gitignore:74]: ah it is below... well after I deleted the plugin reference again this came out :-(

Is nobody of you guys using the plugin for issue tracking?

File: src/main/java/org/spout/engine/command/BanNestedCommands.java [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+
+		// trigger PlayerBanKickEvent
+		PlayerBanKickEvent playerBanKickEvent = eventManager.callEvent(new PlayerBanKickEvent(player, BanChangeEvent.BanType.PLAYER,message.asString()));
+		if (playerBanKickEvent.isCancelled()){
+			source.sendMessage("Ban Player was cancelled by a plugin");
+			return;
+		}
+		message = new ChatArguments(playerBanKickEvent.getMessage());
+		// check if the player is already banned
+		Boolean isAlreadyBanned =  server.getBanManager().isBanned(playerName);
```
Comment (macalinao) [inline src/main/java/org/spout/engine/command/BanNestedCommands.java]: Is there any reason to use the `Boolean` wrapper class here? :squirrel:

Comment (olliestanley) [inline src/main/java/org/spout/engine/command/BanNestedCommands.java]: :squirrel: is the official GitHub sarcasm smiley.

File: src/main/java/org/spout/engine/command/BanNestedCommands.java [hunk 7]
```diff
@@ -0,0 +1,220 @@
...
+
+		// trigger PlayerBanKickEvent
+		PlayerBanKickEvent playerBanKickEvent = eventManager.callEvent(new PlayerBanKickEvent(player, BanChangeEvent.BanType.WHITELIST,message.asString()));
+		if (playerBanKickEvent.isCancelled()){
+			source.sendMessage("Ban Whitelist was cancelled by a plugin");
+			return;
+		}
+		message = new ChatArguments(playerBanKickEvent.getMessage());
+		// check if the player is already whitelisted
+		Boolean isAlreadyWhitelisted =  server.getBanManager().isWhitelisted(playerName);
```
Comment (olliestanley) [inline src/main/java/org/spout/engine/command/BanNestedCommands.java]: Also Boolean is used here, same as @simplyianm's comment :P

**Comment by dredhorse** [comment]
hmm.. why can't I see the comments you made in the diff?

File: src/main/java/org/spout/engine/command/BanNestedCommands.java (hunk 6 above)
Comment (dredhorse) [inline src/main/java/org/spout/engine/command/BanNestedCommands.java]: so what's so negative about it? I need that value several times and I didn't always wanted to write server.getBanManager().isBanned(playerName) instead.

**Comment by olliestanley** [comment]
They changed it, click in the diff tab ;)

**Comment by dredhorse** [comment]
yeah, just found that

File: src/main/java/org/spout/engine/command/BanNestedCommands.java (hunk 6 above)
Comment (stephenwilliams) [inline src/main/java/org/spout/engine/command/BanNestedCommands.java]: @dredhorse @simplyianm is saying you should use `boolean` rather than `Boolean`

Comment (dredhorse) [inline src/main/java/org/spout/engine/command/BanNestedCommands.java]: ah thanks, changed that and also made 2 other changes:

a) moved the if ..... return statements more to the front
b) added an if (player !=null) to the kickbanevent for ban ip

I guess I will push again tomorrow

**Comment by dredhorse** [comment]
hmm now the diffs are back, man... I also really don't like what they did with the new notifications.. have to figure out how to disable notificiations via mail for spout etc..

**Comment by dredhorse** [comment]
forget this PULL... I will redo it again after I got feedback if the general idea is correct.

The pull will not merge anymore and the pull from Windwaker and Sleaker made a lot of the stuff requiring a full rewrite never the less.

so ANY comments about the CODE.... not about spaces and stuff like license formatting?

Also it would be cool if the Spout Team would include .bak and the atlassian plugin into it's .gitignore list

**Comment by lukespragg** [comment]
What Atlassian plugin? Why would .bak need to be included?

**Comment by dredhorse** [comment]
.bak because you don't need it in the repo or?  ... git starts notepad++ for me, which creates .bak files ;-(

https://marketplace.atlassian.com/7627 this one IIRC... it creates a atlassian-ide-plugin.xml file in the project root with idea :-(

**Comment by lukespragg** [comment]
I'll get them added to our projects.

**Comment by dredhorse** [comment]
thanks.

**Comment by dredhorse** [comment]
oh well as the access manager also makes changes to this PR I will close it.

I WOULD REALLY REALLY HOPE that some people will implement the correct events and also remove the duplicate handling of whitelist, ban etc from Spout Config and the old BanManager.

I have invested to much time already into this PR to redo it again and again without it being getting pulled.

**Pull request closed without merging**